src/UserSpaceInstrumentation/include/UserSpaceInstrumentation/ExecuteInProcess.h

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef USER_SPACE_INSTRUMENTATION_EXECUTE_IN_PROCESS_H_
+#define USER_SPACE_INSTRUMENTATION_EXECUTE_IN_PROCESS_H_
+
+#include <sys/types.h>
+
+#include <cstdint>
+#include <string_view>
+
+#include "OrbitBase/Result.h"
+
+namespace orbit_user_space_instrumentation {
+
+// Attach to process stopping the execution, inject `library_path` into process `pid` and execute
+// `function` handing over the paramters. `function` can be any function taking up to six integer
+// parameter and might return an integer parmeter. Finally the process is continued. The library
+// stays loaded until the process finishes.
+// TODO: ... or until we implement something that calls dlclose.
+[[nodiscard]] ErrorMessageOr<uint64_t> ExecuteInProcess(pid_t pid, std::string_view library_path,
```

Comment:
Adjusted according to comment: We now expose the functions for attaching / detaching to a process, the dlopen,dlsym,dlclose wrappers and `ExecuteInProcess`.
This has the additional advantage the the caller can call dlclose at the end leaving the tracce in a cleaned up state after detaching.

I'll remove `FindFunctionAddress` and `ExecuteMachineCode` from the InjectLibraryInTracee.h if this is the way we want to keep it otherwise.

Reply:
The PR is getting a bit convoluted due to all the files being moved - let's chat briefly this evening to go through this and see if this is the solution.